src/javascript/Validation/validation.js

```diff
@@ -55,6 +55,27 @@ const dateFieldValidation = (values, field) => {
     return respectEachConstraint ? undefined : error;
 };
 
+const patternFieldValidation = (values, field) => {
+    const error = 'invalidPattern';
+
+    const constraints = field.valueConstraints && field.valueConstraints.map(constraint => constraint.value.string);
+
+    if (constraints && constraints.length > 0 && field.requiredType === 'STRING') {
+        const fieldValues = field.multiple ? values[field.name] : [values[field.name]];
+
+        // If one pattern is invalid, error!
+        if (fieldValues.some(value =>
```

Comment:
It is not one fieldValue that needs to match the constraint.
it is each field value that needs to match one of the constraint. 
We also need test for this use cases:
- several constraint with no match
- several constraint with only one match
- several values with one that does not match any constraints
- several values with all match of the constraints